homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,406 @@
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE_RANGE,
+)
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    ATTR_MODE,
+    ATTR_NAME,
+    CONF_COMMAND_OFF,
+    CONF_COMMAND_ON,
+    CONF_HOST,
+    CONF_IP_ADDRESS,
+    EVENT_HOMEASSISTANT_STOP,
+    PRECISION_TENTHS,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    ATTR_OVERRIDE_ALLOWED,
+    ATTR_TARGET_ID,
+    ATTR_TARGET_TYPE,
+    ATTR_TEMP_COMFORT_C,
+    ATTR_TEMP_ECO_C,
+    DOMAIN,
+    HUB,
+)
+
+SUPPORT_FLAGS = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE_RANGE
+
+PRESET_MODES = [PRESET_NONE, PRESET_COMFORT, PRESET_ECO, PRESET_AWAY]
+
+HVAC_MODES = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+HVAC_MODES_WITHOUT_OFF = [HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+
+MIN_TEMPERATURE = 7
+MAX_TEMPERATURE = 40
+
+_LOGGER = logging.getLogger(__name__)
+
+_ZONE_NORMAL_WEEK_LIST_SCHEMA = vol.Schema({cv.string: cv.string})
+
+# Legacy configuration to import from HACS version.
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_IP_ADDRESS, default="discover"): cv.string,
+        vol.Optional(CONF_COMMAND_OFF, default=""): cv.string,
+        vol.Optional(CONF_COMMAND_ON, default={}): _ZONE_NORMAL_WEEK_LIST_SCHEMA,
+    }
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config, async_add_entities, discovery_info=None
+) -> bool:
+    """Import Nobø Ecohub configuration from YAML."""
+
+    _LOGGER.warning(
+        "Loading Nobø Ecohub via platform setup is depreciated; Please remove it from your configuration"
+    )
+
+    # Only import if we haven't before.
+    config_entry = _async_find_matching_config_entry(hass)
+    if not config_entry:
+        _LOGGER.warning("Importing Nobø Ecohub from configuration.yaml")
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data=dict(config),
+            )
+        )
+        return True
+
+    return True
+
+
+@callback
+def _async_find_matching_config_entry(hass):
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.source == SOURCE_IMPORT:
+            return entry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_devices
+) -> bool:
+    """Set up the Nobø Ecohub platform from UI configuration."""
+
+    # Setup connection with hub
+    hub = hass.data[DOMAIN][config_entry.entry_id][HUB]
+    await hub.start()
```

Comment:
Should this have been started in the integration setup instead?

Reply:
Yes. That's symmetric with unload. Done.